packages/backend/src/api/endpoints/OrganizationApi.ts

```diff
@@ -285,4 +309,57 @@ export class OrganizationAPI extends AbstractAPI {
       },
     });
   }
+
+  public async getOrganizationDomainList(params: GetOrganizationDomainListParams) {
+    const { organizationId, limit, offset } = params;
+    this.requireId(organizationId);
+
+    return this.request<PaginatedResourceResponse<OrganizationDomain[]>>({
+      method: 'GET',
+      path: joinPaths(basePath, organizationId, 'domains'),
+      queryParams: { limit, offset },
+    });
+  }
+
+  public async createOrganizationDomain(params: CreateOrganizationDomainParams) {
+    const { organizationId, name, enrollmentMode, verified = true } = params;
+    this.requireId(organizationId);
+
+    return this.request<OrganizationDomain>({
+      method: 'POST',
+      path: joinPaths(basePath, organizationId, 'domains'),
+      bodyParams: {
+        name,
+        enrollmentMode,
+        verified,
+      },
+    });
+  }
+
+  public async updateOrganizationDomain(params: UpdateOrganizationDomainParams) {
+    const { organizationId, domainId, enrollmentMode, name, verified } = params;
+    this.requireId(organizationId);
+    this.requireId(domainId);
+
+    return this.request<OrganizationDomain>({
+      method: 'PATCH',
+      path: joinPaths(basePath, organizationId, 'domains', domainId),
+      bodyParams: {
+        ...(enrollmentMode && { enrollment_mode: enrollmentMode }),
+        ...(name && { name }),
+        ...(verified !== undefined && { verified }),
+      },
+    });
```

Comment:
```suggestion
      bodyParams: params 
```

We're already going to filter out `undefined` params here.

Reply:
Oh, gotcha!